### ISSUE TITLE
[Feature]: Resizable component (Splitpane)

### ISSUE THREAD

**Issue by wasimTQ** [radix-vue/shadcn-vue#238]
### Describe the feature

- Create a resizable component using the [SplitPane](https://github.com/antoniandre/splitpanes) library

Part of #220 

### Additional information

- [X] I intend to submit a PR for this feature.
- [ ] I have already implemented and/or tested this feature.

**Comment by sadeghbarati** [comment]
a years ago 👽😪 

Maybe we could use [`@ark-ui/vue` Splitter ](https://ark-ui.com/docs/components/splitter) If it's headless and tree-shakable?

I saw some projects that were based on the main shadcn-ui uses [Material UI Base UI](https://mui.com/base-ui/) for some components that not yet existed in shadcn-ui

O god why Material UI has Base UI but Vuetify hasn't 😬 


---

Anthony Fu worked on the same thing with SplitPanes but he disabled the Splitpanes Trigger with a CSS workaround

https://www.youtube.com/live/IJ27LUb6BAo?si=dUuEH2nGboEIidmB&t=1424

**Comment by wasimTQ** [comment]
Splitpanes has more features than any other projects tho.

But the problem is it's not currently maintained.

I thought maybe we can fork it.

**Comment by wasimTQ** [comment]
I'll play with both this and splitpanes to see which one fits best in a project and which one has more features.

**Comment by wasimTQ** [comment]
@sadeghbarati I was able to get the functionality ready with `splitpanes` but with `ark-ui/vue` not so much luck.

You can check it out here `https://github.com/wasimTQ/shadcn-vue/tree/feature/resizable/apps/www/src/lib/registry/default/ui/resizable`.

In the branch `feature/resizable`

**Comment by wasimTQ** [comment]
@sadeghbarati If possible can you play around with `ark-ui` splitter component. And tell me your thoughts. As far as I worked with it, I couldn't get the exact implementations as the docs

**Comment by sadeghbarati** [comment]
Go with [SplitPane](https://github.com/antoniandre/splitpanes) then if you think that is better?

I'll check your branch, Thanks

**Comment by wasimTQ** [comment]
Problem is Splitpane don't have typescript support. Other than that, it's pretty good for our usecase

**Comment by sadeghbarati** [comment]
Let's fork it and build it with TypeScript support, are you in? I can help

**Comment by wasimTQ** [comment]
I'm in. Let's do it.

**Comment by wasimTQ** [comment]
Shall we merge it as it is for now? I can make all the changes look like the shadcn's resizable

**Comment by sadeghbarati** [comment]
Why not both 🤪 idk

- your ark ui work

- forked splitpanes or splitpanes with @types/splitpanes


Btw if we want to include both, we have to discuss with other maintainers

**Comment by wasimTQ** [comment]
@sadeghbarati Let's discuss whether we should merge it with the current implementation that is without Typescript support for the package.

We already added custom types for props. So I guess that should be enough already.

The `@types/splitpanes` package is not compatible with version `3.5.*`. But it's not much of an issue now.

I'll fork the repo and add typescript support and some additional functionality like accessibility that's not built in.

How it sounds?

**Comment by sadeghbarati** [comment]
@wasimTQ 

There is no rush just here to say if you fork the project You can add me

I can take `Bundling Task` with `unbuild` or `rollup` or `bunchee` or `tsup` or any other good bundlers

**Comment by wasimTQ** [comment]
@sadeghbarati I was little busy with my job. So I couldn't focus on it more. I did clone the project and saw how I can improve it. 

The code is currently written using options API. And I want to change it to composition api and cleanup some code for better readability. And change the documentation to vite.

It'll take some time. I'll add you to the repo once I forked it.

In the meantime, let me know if I can help with any other components. Maybe small fix works.
